pkg/network/ebpf/c/protocols/tls/tls.h

```diff
@@ -1,132 +1,459 @@
+        }
+    } else {
+        // ServerHello
+        // The selected_version field is 2 bytes
+        if (*offset + PROTOCOL_VERSION_LENGTH > data_end) {
+            return false;
+        }
+
+        // Read selected version (2 bytes)
+        __u16 selected_version;
+        if (bpf_skb_load_bytes(skb, *offset, &selected_version, PROTOCOL_VERSION_LENGTH) < 0) {
+            return false;
+        }
+        selected_version = bpf_ntohs(selected_version);
+        *offset += PROTOCOL_VERSION_LENGTH;
+
+        tags->chosen_version = selected_version;
+    }
+
+    return true;
 }
 
-// is_tls_handshake checks if the given TLS message header is a valid TLS
-// handshake message. The message is considered valid if:
-// - The type matches CLIENT_HELLO or SERVER_HELLO
-// - The version is a known SSL/TLS version
-static __always_inline bool is_tls_handshake(tls_record_header_t *hdr, const char *buf, __u32 buf_size) {
-    // Checking the buffer size contains at least the size of the tls record header and the tls hello message header.
-    if (sizeof(tls_record_header_t) + sizeof(tls_hello_message_t) > buf_size) {
+// parse_tls_extensions parses TLS extensions and looks for the supported_versions extension
+static __always_inline bool parse_tls_extensions(struct __sk_buff *skb, __u64 *offset, __u32 data_end, __u64 extensions_end, tls_info_t *tags, bool is_client_hello) {
+    __u16 extension_type;
+    __u16 extension_length;
+
+    #pragma unroll(MAX_EXTENSIONS)
+    for (int i = 0; i < MAX_EXTENSIONS; i++) {
+        if (*offset + MIN_EXTENSION_HEADER_LENGTH > extensions_end) {
+            break;
+        }
+
+        // Read Extension Type (2 bytes)
+        if (bpf_skb_load_bytes(skb, *offset, &extension_type, EXTENSION_TYPE_LENGTH) < 0) {
+            return false;
+        }
+        extension_type = bpf_ntohs(extension_type);
+        *offset += EXTENSION_TYPE_LENGTH;
+
+        // Read Extension Length (2 bytes)
+        if (bpf_skb_load_bytes(skb, *offset, &extension_length, EXTENSION_LENGTH_FIELD) < 0) {
+            return false;
+        }
+        extension_length = bpf_ntohs(extension_length);
+        *offset += EXTENSION_LENGTH_FIELD;
+
+        if (*offset + extension_length > data_end || *offset + extension_length > extensions_end) {
+            return false;
+        }
+
+        if (extension_type == SUPPORTED_VERSIONS_EXTENSION) {
+            if (!parse_supported_versions_extension(skb, offset, data_end, extensions_end, tags, is_client_hello)) {
+                return false;
+            }
+        } else {
+            // Skip other extensions
+            *offset += extension_length;
+        }
+
+        if (*offset >= extensions_end) {
+            break;
+        }
+    }
+
+    return true;
+}
+
+// parse_client_hello parses the ClientHello message and populates tags
+static __always_inline bool parse_client_hello(struct __sk_buff *skb, __u64 offset, __u32 data_end, tls_info_t *tags) {
```

Comment:
please add a link to the client hello message so people can refer to the decoding logic here